datasets/preprocess_utils.py

```diff
@@ -243,3 +243,181 @@ def bin_stream_hilmi24(
+                 8.30152817e-04, -3.57980758e-04]
+
+        fudgeShift = 2.4e-3 # Additive shift in peak (mag)
+    else:
+        raise Exception("Unrecognized Gaia version: %s"%version)
+
+    i10g,i10r,i10i,i10z = pars[0:4]
+    kg,kr,ki,kz = pars[4:8]
+    r1,r2 = pars[8:10]
+    desFlux = (kg * fluxg * (1.0 + fnuPrime[:, 0] * i10g) +
+               kr * fluxr * (1.0 + fnuPrime[:, 1] * i10r) +
+               ki * fluxi * (1.0 + fnuPrime[:, 2] * i10i) +
+               kz * fluxz * (1.0 + fnuPrime[:, 3] * i10z))
+    mGDES = -2.5 * np.log10(desFlux)
+    rMag = -2.5 * np.log10(fluxr)
+    mGDES += r1 * (rMag - 17.0) + r2 * (rMag - 17.0)**2.
+    mGDES += fudgeShift # Peak shift
+    return mGDES
+
+def get_decam_g_r_i_z(iso_name = 'Dotter', iso_age = 11.5, iso_metallicity = 0.00016, iso_distance_modulus = 16.807, iso_stellar_mass = 2e4):
+    """Get simulated DECam g, r, i, z magnitudes from the AAU isochrone"""
+    
+    # Simulated AAU isochrone for g and r bands 
+    iso_g_r = isochrone.factory(name=iso_name,
+                            age=iso_age,  # Gyr
+                            metallicity=iso_metallicity,  # Approximate Z value for the stream metallicity
+                            distance_modulus=iso_distance_modulus,  # Average distance modulus from the stream properties
+                            )
+
+    # Simulated AAU isochrone for i and z bands
+    iso_i_z = isochrone.factory(name=iso_name,
+                            age=iso_age,  # Gyr
+                            metallicity=iso_metallicity,  # Approximate Z value for the stream metallicity
+                            distance_modulus=iso_distance_modulus,  # Average distance modulus from the stream properties
+                            band_1 = 'i',
+                            band_2 = 'z'
+                            )
+
+    decam_g, decam_r = iso_g_r.simulate(stellar_mass=iso_stellar_mass)
+    decam_i, decam_z = iso_i_z.simulate(stellar_mass=iso_stellar_mass)
+
+    return decam_g, decam_r, decam_i, decam_z
+
+def get_gaia_g(decam_g, decam_r, decam_i, decam_z):
+    """Get simulated Gaia G band magnitudes from the AAU isochrone"""
+
+    gaia_g = gaia_transform(decam_g, decam_r, decam_i, decam_z, version='edr3')
+
+    filtered_gaia_g = []
+    
+    color = decam_g - decam_r + 0.04
+    min_color = 0.2
+    max_color = max(color)
+
+    min_mag = 14.875
+    max_mag = 19.512
+
+    for c, m, j in zip(color, decam_r, gaia_g):
+            if min_color <= c <= max_color and min_mag <= m <= max_mag:
+                    filtered_gaia_g.append(j)
+    return filtered_gaia_g
+
+def get_pm_uncertainties(gaia_g):
+    """Get proper motion uncertainties for the simulated stars"""
+    
+    # Calculate proper motion uncertainties for Gaia DR3
+    pmra_unc, pmdec_unc = proper_motion_uncertainty(gaia_g, release='dr3')
+
+    pmra_err = pmra_unc/1000
+    pmdec_err = pmdec_unc/1000
+        
+    return pmra_err, pmdec_err
+
+def add_uncertainty(
```

Comment:
- With the way this function currently works, if the input order is different then the uncertainty will be mixed up. 
- If we remove observables (e.g., vr), then the dimension of the feat array will be different, in these cases, the code will crash. For example, removing `vr` without removing `dist` will crash the code because `feat[:, 4]` no longer exists.

Reply:
Added a new input (feat_list) so the code only adds uncertainties if the feature is found in feat_list